contracts/contracts/UmbraRouter.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.7.6;
+
+contract UmbraRouter {
+  address internal constant umbra = 0xFb2dc580Eed955B528407b4d36FfaFe3da685401;
+
+  /**
+   * @notice Batch send ETH and token payments to a stealth address
+   * @param _receivers Stealth addresses receiving the payment
+   * @param _tokenAddrs Addresses of the ERC20 tokens being sent
+   * @param _tollCommitments Exact toll the sender is paying; should equal contract toll;
+   * the committment is used to prevent frontrunning attacks by the owner;
+   * see https://github.com/ScopeLift/umbra-protocol/issues/54 for more information
+   * @param _pkxes X-coordinate of the ephemeral public key used to encrypt the payload
+   * @param _ciphertexts Encrypted entropy (used to generated the stealth address) and payload extension
+   */
+  function batchSend(
+    address payable[] calldata _receivers,
+    address[] calldata _tokenAddrs,
+    uint256[] calldata _amounts,
+    uint256[] calldata _tollCommitments,
+    bytes32[] calldata _pkxes,
+    bytes32[] calldata _ciphertexts
```

Comment:
Let's define a struct to clean things up and avoid having to verify that all array lengths are equal. We can have `SendEth` struct, a second `SendToken` struct, then a single `tollCommitment` argument (it will be the same for each ETH send, so no need to duplicate that). Then you can have the following function signatures:

```solidity
function batchSendEth(uint256 _tollCommitment, SendEth[] _params);
function batchSendTokens(SendToken[] _params);
function batchSend(uint256 _tollCommitment, SendEth[] _ethParams, SendToken[] _tokenParams);
```

Reply:
You can even just name all three methods `batchSend`, I don't have much preference on which we use

Reply:
@apbendi Any preference? I'm guessing you'll prefer the current approach of separate names. If so just mark this resolved